packages/flutter/lib/src/material/menu_bar.dart

```diff
@@ -0,0 +1,3256 @@
+  const _MenuBarControllerMarker({
+    required this.controller,
+    required super.child,
+  });
+
+  final _MenuBarController controller;
+
+  @override
+  bool updateShouldNotify(covariant _MenuBarControllerMarker oldWidget) {
+    return controller != oldWidget.controller;
+  }
+}
+
+// A widget used as the main widget for the overlay entry in the
+// _MenuBarController. Since the overlay is a Stack, this widget produces a
+// Positioned widget that fills the overlay, containing its own Stack to arrange
+// the menus with. Positioning of the top level submenus is relative to the
+// position of the menu buttons.
+class _MenuStack extends StatelessWidget {
+  const _MenuStack(this.controller);
+
+  final _MenuBarController controller;
+
+  @override
+  Widget build(BuildContext context) {
+    return Positioned.fill(
+      child: FocusScope(
+        node: controller.overlayScope,
+        child: Actions(
+          actions: <Type, Action<Intent>>{
+            NextFocusIntent: _MenuNextFocusAction(controller: controller),
+            PreviousFocusIntent: _MenuPreviousFocusAction(controller: controller),
+            DirectionalFocusIntent: _MenuDirectionalFocusAction(
+              controller: controller,
+            ),
+            DismissIntent: _MenuDismissAction(controller: controller),
+            VoidCallbackIntent: VoidCallbackAction(),
+          },
+          child: Shortcuts(
+            // These are here to make sure that these override any shortcut
+            // bindings from the menu items when a menu is open. If someone
+            // wants to bind an arrow or tab to a menu item, it would otherwise
+            // override the default traversal keys. We want their shortcuts to
+            // apply everywhere but override these in the menu itself, since
+            // there we have to be able to traverse the menus.
+            shortcuts: _kMenuTraversalShortcuts,
+            child: _MenuBarControllerMarker(
+              controller: controller,
+              child: Stack(
+                children: <Widget>[
+                  if (controller.openMenus.isNotEmpty)
+                    ...controller.openMenus.where((_MenuNode node) => node.menuBuilder != null).map<Widget>(
+                      (_MenuNode node) {
+                        return Builder(
+                          key: ValueKey<_MenuNode>(node),
+                          builder: node.menuBuilder!,
+                        );
+                      },
+                    ),
+                ],
+              ),
+            ),
+          ),
+        ),
+      ),
+    );
+  }
+}
+
+class _MenuDismissAction extends DismissAction {
+  _MenuDismissAction({required this.controller});
+
+  final _MenuBarController controller;
+
+  @override
+  bool isEnabled(DismissIntent intent) {
+    return controller.openMenu != null;
+  }
+
+  @override
+  Object? invoke(DismissIntent intent) {
+    controller.closeAll();
+    return null;
```

Comment:
I _think_ you can just change the return type of the override here to `void` and remove the return here.

Reply:
ooh, fancy. Done.